Model/Api/DiscountCodeValidation.php

```diff
@@ -446,6 +446,8 @@ public function validate()
                     ->save();
             }
 
+            $this->eventsForThirdPartyModules->dispatchEvent("applyIdMeToCustomerSession", $parentQuote);
```

Comment:
This event should have a generic name, related to its place and/or functionality in the Bolt code. So we could add listeners to it for other purposes in the future. There's currently only one listener in the listeners array you added below, but they may be more of them eventually.

Reply:
I've updated the code